### PR TITLE
UTOPIA-1366: Refactor Submission behaviour

### PR DESCRIPTION
**Pull request by BradyMitch**
## 🎯 Summary

  
[UTOPIA-1366](https://apps.itsm.gov.bc.ca/jira/browse/UTOPIA-1366)

- Separated util functions for form validation.
- Validation now happens on status change from dropdown.
- Cleaned up code.

## 🔰 Checklist

- [x] I have read and agree with the following checklist and am following the guidelines in our [Code of Conduct](/CODE_OF_CONDUCT.md) document.

> - I have performed a self-review of my code.
> - I have commented my code, particularly in hard-to-understand areas.
> - I have made corresponding changes to the documentation where required.
> - I have tested my changes to the best of my ability.
> - I have consulted with the team if introducing a new dependency.
> - My changes generate no new warnings.

### CODE REVIEW

File: src/frontend/src/pages/PIAForm/index.tsx [hunk 1]
```diff
@@ -643,32 +645,34 @@ const PIAFormPage = () => {
   //
   // Form Submission Handler
   //
-  const handleSubmit = (event: any) => {
+  const handleSubmit = (event: React.FormEvent) => {
     event.preventDefault();
-    // if the user is in pia intake tab, then show submit pia intake; else submit full form
+
+    // Determine which modal to show based on `pia` status.
+    const modalMapping = {
```
Comment (antsand) [inline src/frontend/src/pages/PIAForm/index.tsx:652]: can this be inside the status list logic?

File: src/frontend/src/pages/PIAForm/utils/validateForm.ts [hunk 2]
```diff
@@ -0,0 +1,57 @@
+import { IPiaForm } from '../../../types/interfaces/pia-form.interface';
+import { PiaValidationMessage } from '..';
+import { SetStateAction } from 'react';
+
+type ValidateFormProps = {
```
Comment (antsand) [inline src/frontend/src/pages/PIAForm/utils/validateForm.ts:5]: do we want to have types defined in a separate file?

**Review by NoorChasib — APPROVED** [review]

**Review by antsand — APPROVED** [review]
minor comments, but overall looks very clean. Great job!!

File: src/frontend/src/pages/PIAForm/utils/validateForm.ts (hunk 2 above)
Comment (BradyMitch) [reply src/frontend/src/pages/PIAForm/utils/validateForm.ts:5]: We could but this is a small file already that I think it should just stay close to its implementation

File: src/frontend/src/pages/PIAForm/index.tsx (hunk 1 above)
Comment (BradyMitch) [reply src/frontend/src/pages/PIAForm/index.tsx:652]: Thanks I'll see what I can do

**Review by NoorChasib — APPROVED** [review]

**Pull request merged**